bin/improver-gradient

```diff
@@ -42,20 +42,22 @@ from improver.utilities.spatial import DifferenceBetweenAdjacentGridSquares
 def main():
     """Load in arguments to calculate the gradient between adjacent grid cells
        and save the output gradient fields."""
-    parser = argparse.ArgumentParser(
-        description=('Read the input field, and calculate the gradient'
-                     ' in x and y directions.'))
-    parser.add_argument('--force', dest='force', default=False,
-                        action='store_true',
-                        help=('If True, ancillaries will be generated '
-                              'even if doing so will overwrite existing '
-                              'files.'))
-    parser.add_argument('input_filepath',
-                        metavar='INPUT_FILE',
-                        help='A path to an input NetCDF file to be processed')
-    parser.add_argument('output_filepath', metavar='OUTPUT_FILE',
-                        help='The output path for the processed NetCDF')
-    args = parser.parse_args()
+
+    cli_specific_arguments = [(['--force'],
```

Comment:
Could we keep the old method for adding CLI specific options, rather than putting them into a dictionary? If I'm understanding this correctly then we could create the instance of the ArgParser using all the centralised options we want, and then add new ones within the CLI using the old parse.add_argument. 
The benefit of this is that we don't need knowledge of this dictionary structure in every CLI. It would also mean the ArgParser doesn't need to support adding specific_arguments and only deals with centralised ones. What do you think?

Reply:
There's nothing in the current implementation which prevents us from doing that - add_argument() still works (ad therefore all of the existing CLIs still work). The specific_arguments thing just allows for us to pass them all in, in one go.

Reply:
Oops.. didn't look at which file this comment was about. I understand the context of your comment a little more now.

If it's specifically about this case, and questioning whether we should be creating all of the dictionaries or just using the old method for all of the existing CLIs, I'm not tied strongly to creating the dictionaries upfront. I don't think creating a new CLI using the old approach would be technical debt, actually. (This was just an example to show how we might be able to use it.) For the work I'd like to do in the future, I think the dictionary method would be nicer, but I'm not advocating that we port all CLIs to this approach going forward.

Reply:
I quite like the new way of presenting the arguments; am happy with either implementation for this PR.